src/Controller/CommentController.php

```diff
@@ -0,0 +1,358 @@
+     *     )
+     * )
+     * @SWG\Response(
+     *     response="403",
+     *     description="Forbidden",
+     *     @SWG\Schema(
+     *     @SWG\Property(property="code", type="integer", example=403),
+     *     @SWG\Property(property="message", type="string", example="Access denied!"),
+     *     )
+     * )
+     * @SWG\Response(
+     *     response="400",
+     *     description="Bad request.",
+     *     @SWG\Schema(
+     *     @SWG\Property(property="code", type="integer", example=400),
+     *     @SWG\Property(property="message", type="string", example="Bad request."),
+     *     )
+     * )
+     * @SWG\Response(
+     *     response=404,
+     *     description="Not found",
+     *     @SWG\Schema(
+     *     @SWG\Property(property="code", type="integer", example=404),
+     *     @SWG\Property(property="message", type="string", example="Not found!"),
+     *     )
+     * )
+     * @param Activity $activity
+     * @param Request $request
+     * @param CommentRepository $commentRepository
+     * @param ValidationErrorSerializer $validationErrorSerializer
+     * @return JsonResponse
+     * @throws ORMException
+     * @throws OptimisticLockException
+     */
+    public function addComment(
+        Activity $activity,
+        Request $request,
+        CommentRepository $commentRepository,
+        ValidationErrorSerializer $validationErrorSerializer
+    ): JsonResponse {
+        $authenticatedUser = $this->getUser();
+        $rights = $this->accessRightsPolicy->canAccessActivity($activity, $authenticatedUser);
+
+        if ($rights === false) {
+            return new JsonResponse([
+                'code' => Response::HTTP_FORBIDDEN,
+                'message' => 'Access denied!'
+            ], Response::HTTP_FORBIDDEN);
+        }
+        $data = $request->getContent();
+
+        /** @var DeserializationContext $context */
+        $context = DeserializationContext::create()->setGroups(array('AddComment'));
+
+        $commentDTO = $this->serializer->deserialize(
+            $data,
+            CommentDTO::class,
+            'json',
+            $context
+        );
+        $errors = $this->validator->validate($commentDTO, null, ['AddComment']);
+
+        if (count($errors) > 0) {
+            return new JsonResponse(
+                [
+                    'code' => Response::HTTP_BAD_REQUEST,
+                    'message' => 'Bad Request',
+                    'errors' => $validationErrorSerializer->serialize($errors)
+                ],
+                Response::HTTP_BAD_REQUEST
+            );
+        }
+        $addComment = $this->transformer->addComment($commentDTO, $activity, $authenticatedUser);
+        $commentRepository->save($addComment);
+        return new JsonResponse(['message' => 'Your comment has successfully added!'], Response::HTTP_OK);
+    }
+
+    /**
+     * Edit comment.
+     * @Rest\Post("/activities/{activityId}/edit_comment/{commentId}", requirements={"id"="\d+"})
```

Comment:
In requirements, you have `id` which is not present in query. In query you have `activityId` and `commentId` which are not present in requirements.